### PR TITLE
chore: upgrade to gradle 8.5

### PR DESCRIPTION
**Pull request by aajtodd**
*Issue #, if available:*

*Description of changes:*
Upgrade to Gradle 8.5

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: build-plugins/src/main/kotlin/aws/sdk/kotlin/gradle/kmp/IdeUtils.kt [hunk 1]
```diff
@@ -10,13 +10,14 @@ import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
...
 import org.jetbrains.kotlin.konan.target.KonanTarget
+import java.util.*
 
 /**
  * Whether Intellij is active or not
  */
 val IDEA_ACTIVE = System.getProperty("idea.active") == "true"
 
-val OS_NAME = System.getProperty("os.name").toLowerCase()
+val OS_NAME = System.getProperty("os.name").lowercase(Locale.getDefault())
```
Comment (ianbotsf) [inline build-plugins/src/main/kotlin/aws/sdk/kotlin/gradle/kmp/IdeUtils.kt:20]: **Nit**: Specifying a locale is optional and can be omitted.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-bin.zip
+networkTimeout=10000
```
Comment (ianbotsf) [inline gradle/wrapper/gradle-wrapper.properties:4]: **Question**: Why this new setting?

**Review by ianbotsf — APPROVED** [review]

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (aajtodd) [reply gradle/wrapper/gradle-wrapper.properties:4]: This is auto updated by `./gradlew wrapper --gradle-version=8.5`

File: build-plugins/src/main/kotlin/aws/sdk/kotlin/gradle/kmp/IdeUtils.kt (hunk 1 above)
Comment (aajtodd) [reply build-plugins/src/main/kotlin/aws/sdk/kotlin/gradle/kmp/IdeUtils.kt:20]: IDE auto correct

**Pull request merged**